### PR TITLE
Use DO's metadata service for public interface

### PR DESCRIPTION
**Pull request by caglar10ur**
Also rename PRIVATEIP to VPNIP.

### CODE REVIEW

File: bootstrap/digitalocean_k8s_centos_7_master.sh [hunk 1]
```diff
@@ -37,8 +37,8 @@ systemctl enable docker
 systemctl enable kubelet.service
 systemctl start docker
 
-PUBLICIP=$(curl ifconfig.me)
-PRIVATEIP=$(ip addr show dev tun0 | awk '/inet / {print $2}' | cut -d"/" -f1)
+PUBLICIP=$(curl -s http://169.254.169.254/metadata/v1/interfaces/public/0/ipv4/address)
+PRIVATEIP=$(curl -s http://169.254.169.254/metadata/v1/interfaces/private/0/ipv4/address)
```
Comment (xmudrii) [inline bootstrap/digitalocean_k8s_centos_7_master.sh]: `PRIVATEIP` is actually VPN's interface IP, not Droplet's Private IP, so we can't use metadata here. We'll have to resort back to `ip`.

File: bootstrap/digitalocean_k8s_ubuntu_16.04_master.sh [hunk 2]
```diff
@@ -23,9 +23,9 @@ apt-get install -y \
 systemctl enable docker
 systemctl start docker
 
-PRIVATEIP=$(ip addr show dev tun0 | awk '/inet / {print $2}' | cut -d"/" -f1)
+PUBLICIP=$(curl -s http://169.254.169.254/metadata/v1/interfaces/public/0/ipv4/address)
+PRIVATEIP=$(curl -s http://169.254.169.254/metadata/v1/interfaces/private/0/ipv4/address)
```
Comment (xmudrii) [inline bootstrap/digitalocean_k8s_ubuntu_16.04_master.sh]: Same here.

**Comment by xmudrii** [comment]
Thanks a lot for your contribution! I've left you some comments that needs to be addressed before we can merge this.

File: bootstrap/digitalocean_k8s_centos_7_master.sh (hunk 1 above)
Comment (xmudrii) [reply bootstrap/digitalocean_k8s_centos_7_master.sh]: Maybe we can also rename this to something more obvious.

Comment (caglar10ur) [reply bootstrap/digitalocean_k8s_centos_7_master.sh]: How about renaming it to VPNIP?

Comment (xmudrii) [reply bootstrap/digitalocean_k8s_centos_7_master.sh]: Great — it will do the job.

**Comment by caglar10ur** [comment]
@xmudrii refreshed the PR

File: bootstrap/digitalocean_k8s_ubuntu_16.04_master.sh [hunk 3]
```diff
@@ -23,15 +23,15 @@ apt-get install -y \
 systemctl enable docker
 systemctl start docker
 
-PRIVATEIP=$(ip addr show dev tun0 | awk '/inet / {print $2}' | cut -d"/" -f1)
-echo $PRIVATEIP > /tmp/.ip
-PUBLICIP=$(curl ifconfig.me)
+PUBLICIP=$(curl -s http://169.254.169.254/metadata/v1/interfaces/public/0/ipv4/address)
+VPNIP=$(curl -s http://169.254.169.254/metadata/v1/interfaces/private/0/ipv4/address)
```
Comment (xmudrii) [inline bootstrap/digitalocean_k8s_ubuntu_16.04_master.sh]: Same problem here, we'll have to replace metadata with `ip`.

**Comment by xmudrii** [comment]
@caglar10ur left you one more comment. Once it's fixed, I'll happily merge the PR.

File: bootstrap/digitalocean_k8s_ubuntu_16.04_master.sh (hunk 3 above)
Comment (caglar10ur) [reply bootstrap/digitalocean_k8s_ubuntu_16.04_master.sh]: Yep, just realized that a second ago #needmorecoffee :) Should be fixed now

**Comment by xmudrii** [comment]
@caglar10ur Thank you so much! LGTM!

**Pull request merged**